proto/Agent.proto

```diff
@@ -26,6 +26,19 @@ enum AgentMessageType {
   DECRYPT_FILE = 20;
   GET_PRIMARY_KEYPAIR = 21;
   UPDATE_SECRET = 22;
+  GET_PEER_INFO = 23;
+  ADD_PEER = 24;
+  CLONE_VAULT = 25;
+  PULL_VAULT = 26;
+  PING_PEER = 27;
+  FIND_PEER = 28;
+  FIND_SOCIAL_PEER = 29;
+  LIST_PEERS = 30;
+  TOGGLE_STEALTH = 31;
+  UPDATE_PEER_INFO = 32;
+  REQUEST_RELAY = 33;
+  REQUEST_PUNCH = 34;
+  SCAN_VAULT_NAMES = 35;
 }
 message AgentMessage {
```

Comment:
So because GRPC is not accepting UDS, and we want to use UDS. Here we are just defining protobuf schemas, and doing the protobuf message exchange manually without GRPC.

The reason to use UDS is simplicity and security. UDS doesn't require a port allocation. It can be namespaced to a specific directory.

However packet sniffing on localhost is only possible with superuser privileges. So I think it is fine for us to use GRPC on the local network. It just means we have autofind a port. This is also possible because we can use `0` and autofind a port for allocation. See how we do this with http servers.

One problem isn't just sniffing. It's that once you are listening on localhost, any other program can try to connect to that port. Is this a problem? Yes it is a problem because any user (non-root) that is not the same user can now run a program that tries to get secrets from the agent.

So to resolve this problem, if we want to use grpc (and future proof it), it means we also need to use use a token system between the pk client and the agent. This also means the pk client is now maintaining state. This might work well since it could mean `pk` could run remotely.

Just to map this out. The `pk` client is one of the parties that is maintaining a "session" to this system. The keynode agent is itself a party. And the vaults themselves are openable and closable. If the `pk` client is interacting with the keynode agent over GRPC. That makes the pk client a very similar program to other agents. So other keynode agents can be thought of as clients to the keynode agent as well. So we could standardize on all of these access logic and authentication session system using OAuth2. This does complicate the pk client a bit more (that's the cost of this idea), but the benefit is that now every inter-agent communication and pkclient to agent communication is standardized and therefore reduces the surface area of complexity.

Reply:
So for this PR we will stick with how its been done atm. But going ahead, we will GRPC for everything here.

Reply:
updated issue: https://github.com/MatrixAI/js-polykey/issues/90 with relevant information